lib/dataset-tools/parsers.ts

```diff
@@ -405,21 +413,36 @@ class DialogueParser extends Stream.Transform {
                 if (key === 'context' && this._keySequence[expect] === 'user')
                     key = 'intermediate_context';
 
+                if (this._ignoreErrors) {
+                    if (key === 'agent' && this._keySequence[expect] === 'context') {
+                        currentTurn.context = currentTurn.user_target;
+                        expect = (expect + 1) % this._keySequence.length;
```

Comment:
what does this do?

Reply:
Basically, sometimes in the recording you have a sequence:
```
U: foo
UT: $failed;
A: Sorry I don't know how to do that yet
...
```
(or similar)

This is malformed because between UT and A you should have C
Those lines "adjust" the context to be:
```
U: foo
UT: $failed;
C: $failed;
A: Sorry I don't know how to do that yet
...
```
And then we can keep going.